include/exiv2/xmp_exiv2.hpp

```diff
@@ -361,7 +361,7 @@ class EXIV2API XmpParser {
 
     @return True if the initialization was successful, else false.
    */
-  static bool initialize(XmpParser::XmpLockFct xmpLockFct = nullptr, void* pLockData = nullptr);
+  static bool initialize(const XmpParser::XmpLockFct& xmpLockFct = nullptr, void* pLockData = nullptr);
```

Comment:
I'm surprised that you can initialize this reference with `nullptr`. Does it create a no-op function?

Reply:
I would think so. There's still overhead in all of the creation and whatnot. Might be better to remove honestly...

Reply:
As far as I can tell, this feature is never used. I.e. `initialize` is always called with zero arguments. The long comment does suggest that somebody has spent quite a bit of time designing it though, so maybe some people who use exiv2 as a library are using this feature.

I'm a bit nervous about changing this code because we don't have any tests for it. I'm concerned that the const reference could cause a dangling reference bug if the caller isn't careful about the lifetime of the `xmpLockFct` argument. Also, I don't think it's worth changing for performance reasons because `initialize` only gets called once. So I'd prefer to leave this code unchanged.